docs/articles/documentation/guides/advanced-guides/speed-up-test-execution.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: docs
+title: Create Faster Tests
+permalink: /documentation/guides/advanced-guides/speed-up-test-execution.html
+---
+
+# Speed Up Test Execution
+
+This article describes ways to decrease the execution time of TestCafe tests. It includes the following sections:
+
+* [Run Tests Concurrently](#run-tests-concurrently)
+* [Run Tests in Headless Browsers](#run-tests-in-headless-browsers)
+* [Use Roles for Login](#use-roles-for-login)
+* [Set Test Speed](#set-test-speed)
+* [Run Tests in Local Browsers](#run-tests-in-local-browsers)
+* [Mock Requests](#mock-requests)
+* [Optimize Your Page Model](#optimize-your-page-model)
+
+## Run Tests Concurrently
+
+Enable concurrent mode to run multiple browser instances simultaneously. This speeds up the testing process at the expense of higher resource consumption.
+
+You can enable concurrency with the [concurrency](../../reference/configuration-file.md#concurrency) configuration file property:
+
+```json
+{
+  "concurrency": 3
+}
+```
+
+In CLI, use the [--concurency](../../reference/command-line-interface.md#-c-n---concurrency-n) option:
+
+```sh
+testcafe --concurency 3 chrome tests/
+```
+
+For more info on concurrency, read [Run Tests Concurrently](../../guides/basic-guides/run-tests.md#run-tests-concurrently).
+
+## Run Tests in Headless Browsers
+
+Headless browsers take less time to initialize because they lack a GUI. They also enable you to run tests in containers and other environments without graphical capabilities.
+
+For more information, see [Test in Headless Mode](../../guides/concepts/browsers.md#test-in-headless-mode)
+
+## Use Roles for Login
+
+Use [Roles](../../guides/advanced-guides/authentication.md#user-roles) to save time on user authentication.
+
+Roles are user log-in routines. When you first enable a `Role`, TestCafe executes the routine and saves the user authentication data that it receives. When you activate the same role later, TestCafe loads the authentication data and skips the log-in process.
+
+When you switch between `Roles`, TestCafe replaces your browser's existing authentication data with the new role's credentials.
```

Comment:
Please, add this this thought:
```js
So, the test speed up is due to authentication action is executed only 1 time.
```

Reply:
This idea is expressed in a previous sentence

> When you activate the same role later, TestCafe loads the authentication data and skips the log-in process.

Reply:
The `Role` is a unusual concept. So, we strictly say why Role usage speed up the tests.

Reply:
I hear you. What about:

> Roles are user log-in routines. When you first enable a `Role`, TestCafe executes the routine and saves the user authentication data. When you activate the same role again, TestCafe loads the authentication data and the log-in happens much faster.